docker.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+
+if [ "${DOCKER_IMAGE_NAME}" = "" ]
+then
+  echo "Docker deployment ignored env DOCKER_IMAGE_NAME has not been provided"
+  exit;
+fi
+
+if [ ! "${DOCKER_PASSWORD}" = "" ]
```

Comment:
why is it checked? there's no else behaviour and if not provided the script just goes on? and `$DOCKER_USERNAME` is not checked?

Reply:
You can run docker.sh in your dev computer without re-login

Reply:
yes but what if you run it not in your dev computer without providing the details?

Reply:
I guess the check could include $DOCKER_USERNAME, $DOCKER_PASSWORD or both

Reply:
if there are no docker credentials at all (params or pre `docker login`), docker push will fail with an unauthorized error

Reply:
yes, but after the build - i'd prefer the script to fail if no username and password are provided, after all this is automation, not a script that we would usually expect to be run locally

Reply:
OK, ignoring, rejecting, or reusing is fine for me.  Let me update to fail if not all params are provided

Reply:
Updated...